### PR TITLE
Add compress_kv for Baichuan2

### PR DESCRIPTION
**Pull request by hxsz1997**
## Description

Add compress_kv for Baichuan2

### 4. How to test?
- [ ] N/A
- [ ] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.
- [ ] Application test
- [ ] Document test
- [ ] ...

### 5. New dependencies

- [ ] New Python dependencies
       - Dependency1 
       - Dependency2
       - ...
- [ ] New Java/Scala dependencies and their license
       - Dependency1 and license1
       - Dependency2 and license2
       - ...

### CODE REVIEW

**Comment by hxsz1997** [comment]
PR validation: https://github.com/intel-analytics/ipex-llm-workflow/actions/runs/10501414480

File: python/llm/src/ipex_llm/transformers/models/baichuan.py [hunk 1]
```diff
@@ -92,7 +257,11 @@ def baichuan_attention_forward_7b(
 
     kv_seq_len = key_states.shape[2]
     if past_key_value is not None:
-        kv_seq_len += past_key_value[0].shape[2]
+        if use_compresskv:
```
Comment (cyita) [inline python/llm/src/ipex_llm/transformers/models/baichuan.py:260]: # [CompressKV]

File: python/llm/src/ipex_llm/transformers/models/baichuan.py [hunk 2]
```diff
@@ -83,6 +246,8 @@ def baichuan_attention_forward_7b(
     bsz, q_len, _ = hidden_states.size()
     device = hidden_states.device
 
+    use_compresskv = isinstance(past_key_value, DynamicCompressCache)
```
Comment (cyita) [inline python/llm/src/ipex_llm/transformers/models/baichuan.py:249]: `# [CompressKV]`

File: python/llm/src/ipex_llm/transformers/models/baichuan.py [hunk 3]
```diff
@@ -130,6 +310,10 @@ def baichuan_attention_forward_7b(
         if use_quantize_kv:
```
Comment (cyita) [inline python/llm/src/ipex_llm/transformers/models/baichuan.py:310]: if use_compresskv:
            attention_mask = get_compresskv_attn_mask(key_states, attention_mask)

if use_quantize_kv:
...
else:
....

**Review by cyita — APPROVED** [review]
LGTM

**Pull request merged**